File: vaarthahub_app/lib/models/newspaper.dart
class Newspaper {
  final int newspaperId;
  final int itemId;
  final String name;
  final String category;
  final String paperType;
  final double basePrice;
  final String? logoUrl;
  final bool isActive;

  Newspaper({
    required this.newspaperId,
    required this.itemId,
    required this.name,
    required this.category,
    required this.paperType,
    required this.basePrice,
    this.logoUrl,
    required this.isActive,
  });

  factory Newspaper.fromJson(Map<String, dynamic> json) {
    return Newspaper(
      newspaperId: json['newspaperId'] ?? 0,
      itemId: json['itemId'] ?? 1,
      name: json['name'] ?? '',
      category: json['category'] ?? '',
      paperType: json['paperType'] ?? '',
      basePrice: (json['basePrice'] ?? 0).toDouble(),
      logoUrl: json['logoUrl'],
      isActive: json['isActive'] ?? true,
    );
  }
}

File: vaarthahub_api/vaarthahub_api/Models/Newspaper.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace vaarthahub_api.Models
{
    public class Newspaper
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int NewspaperId { get; set; }

        public int ItemId { get; set; } = 1;

        public string ItemType { get; set; } = "Newspaper";

        public string Name { get; set; } = string.Empty;

        public string Category { get; set; } = string.Empty;

        public string PaperType { get; set; } = string.Empty;

        [Column(TypeName = "decimal(6, 2)")]
        public decimal BasePrice { get; set; }

        public string? LogoUrl { get; set; }

        public bool IsActive { get; set; } = true;
    }
}
